### PR TITLE
Fix mypy error with dataclasses

### PR DESCRIPTION
**Pull request by olethanh**
Had do ignore since mypy expect a DataClassIstance that don't exist

### CODE REVIEW

File: src/aleph/vm/utils/__init__.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def to_json(o: Any):
     elif hasattr(o, "dict"):  # Pydantic
         return o.dict()
     elif is_dataclass(o):
-        return dataclass_as_dict(o)
+        return dataclass_as_dict(o)  # type: ignore
```
Comment (hoh) [inline src/aleph/vm/utils/__init__.py:78]: What is the error and isn't there a more elegant way to solve the issue ?
`# type: ignore` looks a bit like "force-ignore-this".

Comment (Psycojoker) [reply src/aleph/vm/utils/__init__.py:78]: And shouldn't the return type be defined for this function?

Comment (olethanh) [reply src/aleph/vm/utils/__init__.py:78]: @hoh That's explained in the PR description why I had to force ignore.

error was
```
src/aleph/vm/utils/__init__.py:78: error: No overload variant of "asdict" matches argument type "type[DataclassInstance]"  [call-overload]
src/aleph/vm/utils/__init__.py:78: note: Possible overload variants:
src/aleph/vm/utils/__init__.py:78: note:     def asdict(obj: DataclassInstance) -> dict[str, Any]
src/aleph/vm/utils/__init__.py:78: note:     def [_T] asdict(obj: DataclassInstance, *, dict_factory: Callable[[list[tuple[str, Any]]], _T]) -> _T
Found 1 error in 1 file (checked 71 source files)
```
DataclassInstance is not something that exist, so for me it looks like a mypy bug.  But if you have another fix I would be happy to accept it.

@Psycojoker which function?

Comment (Psycojoker) [reply src/aleph/vm/utils/__init__.py:78]: @olethanh github doesn't allow me to comment on any line but the exact function you are editing

image

Comment (olethanh) [reply src/aleph/vm/utils/__init__.py:78]: yeah it probably should but it's totally outside the scope of this PR

Comment (Psycojoker) [reply src/aleph/vm/utils/__init__.py:78]: Ended up doing it here https://github.com/aleph-im/aleph-vm/pull/704

**Comment by Psycojoker** [comment]
I'm a bit confused, I can't reproduce it locally, is it using "hatch run lint:typing"? Or with a specific version of mypy or on another code base or PR?

File: src/aleph/vm/utils/__init__.py (hunk 1 above)
Comment (hoh) [reply src/aleph/vm/utils/__init__.py:78]: I merged #704

**Comment by olethanh** [comment]
It was one the solana branch, which added the dep : `solathon==1.0.2`

**Pull request closed without merging**